### PR TITLE
Emulate wrfsbase instruction

### PR DESCRIPTION
**Pull request by anakrish**
Emulate wrfsbase instruction.

### CODE REVIEW

File: enclave/core/sgx/exception.c [hunk 1]
```diff
@@ -219,8 +219,61 @@ int _emulate_illegal_instruction(sgx_ssa_gpr_t* ssa_gpr)
...
+        };
+
+        uint8_t last_byte = ((uint8_t*)ssa_gpr->rip)[4];
+        uint8_t idx = last_byte - 0xd0;
+        if (idx >= sizeof(regs) / sizeof(regs[0]))
+            return -1;
+
+        ssa_gpr->fs_base = regs[idx];
+        ssa_gpr->rip += 5;
+        return 0;
```
Comment (jhand2) [inline enclave/core/sgx/exception.c:276]: Nit: this code is the same as the above code right? Can this be factored into `oe_emulate_wrfsbase()`. Also would rdfsbase also need to be emulated?

Feel free to ignore until we decide if the approach will be accepted or not.

Comment (anakrish) [reply enclave/core/sgx/exception.c:276]: The registers are different. If found this to be more readable than if-else.

Comment (anakrish) [reply enclave/core/sgx/exception.c:276]: We could emulate `rdfsbase` if needed. It is possible.

Comment (jhand2) [reply enclave/core/sgx/exception.c:276]: Oh I see. That's fine then.

**Comment by anakrish** [comment]
bors try

**Comment by jhand2** [comment]
Test failures are expectd

**Pull request merged**